Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SpatialLocalizer.cs

```diff
@@ -32,53 +29,68 @@ public abstract class SpatialLocalizer : MonoBehaviour
         /// <summary>
         /// The type name of this object instance.
         /// </summary>
-        public string TypeName => typeName ?? (typeName = GetType().Name);
+        private string TypeName => typeName ?? (typeName = GetType().Name);
 
         /// <summary>
         /// Helper method for logging troubleshooting information.
         /// </summary>
         /// <param name="message">Message to log.</param>
         /// <param name="token">The <see cref="Guid"/> token representing the request.</param>
-        protected void DebugLog(string message, Guid token)
+        protected void DebugLog(string message)
         {
             if (debugLogging)
             {
-                Debug.Log($"{TypeName} - {token}: {message}");
+                Debug.Log($"{TypeName}: {message}");
             }
         }
 
-        /// <summary>
-        /// Start a new spatial localization session, and return a token for that session.
-        /// </summary>
-        /// <param name="actAsHost">If true, this localizer will operate under the assumption that it is hosting the spatial coordinate system.</param>
-        /// <param name="cancellationToken">The cancellation token to cancel this async operation.</param>
-        /// <returns>The token representing this session, should be used for subsequent calls.</returns>
-        internal abstract Task<Guid> InitializeAsync(bool actAsHost, CancellationToken cancellationToken);
+        protected virtual void Start()
+        {
+            if (!SpatialCoordinateSystemManager.IsInitialized)
+            {
+                Debug.LogError($"{TypeName} - Failed to register spatial localizer with the {nameof(SpatialCoordinateSystemManager)}");
+                return;
+            }
 
-        /// <summary>
-        /// Deinitializes the session.
-        /// </summary>
-        /// <param name="actAsHost">If true, this localizer will operate under the assumption that it is hosting the spatial coordinate system.</param>
-        /// <param name="token">The token representing the session.</param>
-        internal abstract void Uninitialize(bool actAsHost, Guid token);
+            SpatialCoordinateSystemManager.Instance.RegisterSpatialLocalizer(this);
+        }
```

Comment:
Does this require an active game object/enabled script for registration? Is our thought that each type of SpatialLocalizer will have a component enabled in the scene in order for registration to work?

Reply:
Yes, some active GameObject is necessary for this script to be present on to register the localizer. I'm open to other ideas for how this could work, but this matches the pattern used for ComponentBroadcasterService registration